script.js

```diff
@@ -1,4 +1,89 @@
+var PlayerOne = [];
+var playerTwo = [];
+
+// Game modes
+var gameModeDiceRoll = "game mode dice roll player 1";
+var gameModeChooseDice = "game mode choose dice";
+
+// Game mode 1 roll dice
+var mode = gameModeDiceRoll;
+
+// generate dice number from 1-6
+var diceRoll = function () {
+  var randomDiceRoll = Math.ceil(Math.random() * 6);
+  return randomDiceRoll;
+};
+// player to get random number
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (mode == gameModeDiceRoll) {
+    playerOne = [diceRoll(), diceRoll()];
+    console.log(playerOne);
+    myOutputValue =
+      "Welcome Player 1." +
+      "<br>" +
+      "You rolled " +
+      playerOne[0] +
+      " for Dice 1 and " +
+      playerOne[1] +
+      " for Dice 2. Please choose the order of the dice.";
+    mode = gameModeChooseDice;
+    return myOutputValue;
+  }
+  // Player 1: Game mode to choosing the number
+  if (mode == gameModeChooseDice) {
+    if (input == "1") {
+      myOutputValue =
+        " Player 1, you chose Dice 1 first. Your number is " +
+        playerOne[0] +
+        playerOne[1] +
+        ". It is now Player 2's turn.";
+    }
+    if (input == "2") {
+      myOutputValue =
+        " Player 1, you chose Dice 2 first. Your number is " +
+        playerOne[1] +
+        playerOne[0] +
+        ". It is now Player 2's turn.";
+      mode = gameModeDiceRoll;
+      return myOutputValue;
+    }
+  }
+
+  // Player 2
+  if (mode == gameModeDiceRoll) {
```

Comment:
I think you've realised that the game doesn't work the way you think it would? One key reason is that the mode name for the first player is the same as the mode name for the second player. This is one area that needs to be fixed, so that we can clearly separate what values are assigned to the first player and what should be assigned to the second player.